### ISSUE TITLE
[meta] Sunset the legacy infrastructure for usage stats

### ISSUE THREAD

**Issue by willdurand** [mozilla/addons-server#14620]
We should identify which CRON tasks should be removed and how we would do that. **Is there anything else we should do?** I guess we'll be able to delete the `update_counts` table as well.

I think the only blocker right now is: https://github.com/mozilla/addons-server/issues/14609. It should land in two weeks and we might as well start using it (by turning the waffle switch on). It should only impact the number of users (`average_daily_users`).

We will need two steps to complete the rollout:

1. give access to beta stats to 100% of our users
2. make beta stats the new and only stats, which should be done by July 9th according to this blog post: https://blog.mozilla.org/addons/2020/06/10/improvements-to-statistics-processing-on-amo/

@diox @bqbn thoughts?

**Comment by diox** [comment]
Note that completing the rollout and sunsetting the legacy infrastructure are two different things - we don't have to remove the old infra right away after making the switch, we can wait a bit to ensure everything works fine.

**Comment by bqbn** [comment]
Ditto. Let's roll out the new thing first, wait for a week or so then come back to this issue.

From Ops side, after we remove the related cron jobs and database tables, we can start the work of moving versioncheck endpoint to cloudfront, and reducing the number of elbs too.

**Comment by diox** [comment]
data-wise, that means we'll be able to remove:
- The entire `update_counts` MySQL table
- All `addons_stats_update_counts*` elasticsearch index & aliases

We'll want to remove all associated code in a prior push to be safe though. The `update_counts_from_file` cron, the `UpdateCount` model, the `UpdateCountIndexer` class, support for those in tasks & views, etc.

**Comment by willdurand** [comment]
> We will need two steps to complete the rollout:
> 
>     1. give access to beta stats to 100% of our users
> 
>     2. make beta stats the new and only stats, which should be done by July 9th according to this blog post: [blog.mozilla.org/addons/2020/06/10/improvements-to-statistics-processing-on-amo](https://blog.mozilla.org/addons/2020/06/10/improvements-to-statistics-processing-on-amo/)

So (1) is done since last Tuesday and the patch for (2) landed yesterday (https://github.com/mozilla/addons-server/issues/14807). As expected, next push will contain lots of changes to stop relying on "update counts"/legacy usage stats (see: https://github.com/mozilla/addons-server/issues/14811).

We'll be able to delete the ES index/aliases after next week's push: https://github.com/mozilla/addons-server/issues/14823.

We'll delete the `update_counts` table the week after next week: https://github.com/mozilla/addons-server/issues/14812.

In summary:

- 2020-06-30: beta stats available to everyone (100% of our users)
- 2020-07-02: deploy minor improvements to the beta stats
- 2020-07-09: make beta stats the default stats, remove code that relies on the ES index and `UpdateCount` model, stop executing the `update_counts_from_file` task 
- 2020-07-16: remove the `update_counts` DB table

Does that work for everyone?

**Comment by diox** [comment]
WFM, although technically #14811 is blocked by https://github.com/mozilla/addons-server/issues/14815 right ? We need to ensure we have something to compute or avoid exposing trending before we remove UpdateCount and the old pipeline.

**Comment by willdurand** [comment]
> although technically #14811 is blocked by #14815 right ? We need to ensure we have something to compute or avoid exposing trending before we remove UpdateCount and the old pipeline.

I thought we decided to live without `hotness` update for a week if #14815 is not solved in next week's tag? The patch for #14811 makes the `deliver_hotness` task a no-op task in the meantime.

**Comment by diox** [comment]
Ah, that works. Carry on.

**Comment by willdurand** [comment]
`deliver_hotness` cron task has been updated and will be deployed this week

**Comment by willdurand** [comment]
> `deliver_hotness` cron task has been updated and will be deployed this week

err, we'll deploy a new version of this task this week (the previous version wasn't efficient so we stopped it).

**Comment by willdurand** [comment]
Looks like we can close this issue.

@bqbn this week, we'll delete the `update_counts` table. is it fine to do it _via_ a migration or should we empty the table beforehand?

**Comment by bqbn** [comment]
If nothing is using that table, let's do it in the post-deployment step.

**Comment by willdurand** [comment]
> If nothing is using that table, let's do it in the post-deployment step. Or put it in the migration, if it doesn't take much time to finish.

there is a migration already, not sure how long it took in -dev. we can probably take a look at the logs.

**Comment by willdurand** [comment]
@bqbn looks like it took no time in dev and stage according to Stackdriver. I guess we're good to run the migration as usual, then?

**Comment by bqbn** [comment]
Yeah, it should be fine. Since we're dropping the table, it shouldn't cause metadata locks anyways.
